clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java

```diff
@@ -285,24 +310,24 @@ private EhcacheEntityResponse invokeLifeCycleOperation(ClientDescriptor clientDe
       case DESTROY_SERVER_STORE:
         destroyServerStore(clientDescriptor, (DestroyServerStore) message);
         break;
+      case CREATE_CLUSTERED_MAP:
+        createConcurrentMap(clientDescriptor, (CreateClusteredMap) message);
```

Comment:
There won't be two server round trips for a single map creation/retrieval. There'll only be one. I believe @ljacomet is referring to the _putIfAbsent_ and subsequent _get_ calls in `ClusteredStateRepository.getPersistentConcurrentMap()` method. Here the _putIfAbsent_ and _get_ happens on a local map *knownMaps* that keeps track of the client-side end points of clustered maps created. It is the `ConcurrentClusteredMap` constructor that is responsible for the creation/retrieval of the clustered map on the server side. The failure of _putIfAbsent_ would mean that there was a clustered map endpoint available already on the client-side which is attached to the server-side map. The success of _putIfAbsent_ implies that a clustered map has been created on *or attached to* the server side and the client-side endpoint to that map is added to the *knownMaps*. In all cases there is only one round trip.

I think the operation name is a bit confusing here. It should have been `CREATE_OR_ATTACH_CLUSTERED_MAP` and not just `CREATE_CLUSTERED_MAP` because that's what actually happens on the server side. There is no 2-step creation and retrieval. Both happens with that single message.